examples/2d/elixir_advection_p4est_non_conforming_flag.jl

```diff
@@ -21,12 +21,12 @@ f4(s) = SVector(s,  1.0 + sin(0.5 * pi * s))
 
 trees_per_dimension = (3, 2)
 
-# Create P4estMesh with 8 x 8 trees and 16 x 16 elements
+# Create P4estMesh with 3 x 2 trees and 6 x 4 elements
 mesh = P4estMesh(trees_per_dimension, polydeg=3,
```

Comment:
Why do you think that it should be a keyword argument?

Reply:
I think it's easier to understand
```julia
mesh = P4estMesh(trees_per_dimension=(3, 2), polydeg=3, ...)
```
than
```julia
mesh = P4estMesh((3, 2), 3, ...)
```
Avoiding the need to pull `trees_per_dimension = (3, 2)` out of the constructor allows us to modify this argument in `trixi_include` even if this wasn't intended by the person mixing the elixir in the first place (and it saves a line - bikeshedding).

Reply:
But I don't have a very strong opinion here. If you and @sloede prefer
```julia
trees_per_dimension = (3, 2)
mesh = P4estMesh(trees_per_dimension, polydeg=3, ...)
```
that's okay for me.

Reply:
@sloede, what do you prefer?

Reply:
I think if the ctor is sufficiently powerful, it would make sense to rely on keyword arguments (with the added benefit that it allows easier overriding via `trixi_include`).

Other than that, I think the ctor interface will need some more iterations (e.g., when we add the capability to turn any `UnstructuredQuadMesh` into a `P4estMesh`) before it is converged, so if it is not 100% thought out yet, I think that's ok for now.